docs/ci/platforms.mdx

```diff
@@ -0,0 +1,143 @@
+In this document we'll outline a few ways for running Patrol UI tests of Flutter
+apps.
+
+Generally, the solutions for running UI tests of mobile apps can be divided into
+2 groups:
+
+- Device labs - platforms that provide access mobile devices in the cloud. You
+  upload an app binary with tests to the device lab, which runs the tests and
+  reports the results back to you.
+
+- Traditional – containers or a VM, either managed or self-hosted. In this
+  approach, you get access to the shell, so everything is possible. You manually
+  script what you want to achieve, which is usually: installing the Android/iOS
+  SDK tools, creating a virtual device, running tests, and collecting results.
+
+There are quite a few solutions in each of these groups, and each is unique, but
+generaly, **device labs trade flexibility for ease of use**. They're a good fit
+for most apps but make certain more complicated scenarios impossible.
+
+An example of a more advanced use case that is not possible to accomplish on
+device labs is scanning a QR code. One of the apps we worked on had this
+feature, and we wanted to test it because it was critical part of the flow. When
+you have access to the shell filesystem (which you do have in the "manual"
+approach, and don't have in the "device lab" approach), you can easily replace
+the scene that is visible in the camera's viewfinder:
+
+This is not possible on device labs.
+
+# Device labs
+
+### Firebase Test Lab
+
+Firebase Test Lab is one of the most popular device labs. It is a good choice
+for most projects.
+
+**Things to keep in mind**
+
+When running on Firebase Test Lab, your app is automatically granted all
+permissions. If you want to test the flow of requesting permissions, you're out
+of luck.
```

Comment:
This is true only for Android. on iOS, permissions are not granted by default, maybe this should be clarified in that paragraph?
Also, the `gcloud alpha` version of gcloud commands adds [a new `--grant-permissions` flag](https://cloud.google.com/sdk/gcloud/reference/alpha/firebase/test/android/run#--grant-permissions) which, when passed the value "none", resets all permissions and allows tests to ask for them.
Example:
```shell
gcloud alpha firebase test android run \
        --type instrumentation \
        --app build/app/outputs/apk/debug/app-debug.apk \
        --test build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk \
        --timeout 1m \
        --grant-permissions none
```

Reply:
Thanks - I didn't know about it. Will fix the docs accordingly.